src/Aspire.Hosting/Dcp/Process/ProcessUtil.cs

```diff
@@ -39,7 +39,7 @@ public static (Task<ProcessResult>, IAsyncDisposable) Run(ProcessSpec processSpe
             process.StartInfo.Environment[key] = value;
         }
 
-        var processEventLock = new object();
+        var startupComplete = new ManualResetEventSlim(false);
```

Comment:
```suggestion
        Reset event to prevent output processing and exited events from running until OnStarted is complete.
        In OnStarted there might be some code that sets up data structures that then are used by these events.
        var startupComplete = new ManualResetEventSlim(false);
```